contracts/OwnedSet.sol

```diff
@@ -26,21 +26,15 @@ contract OwnedSet is Owned, ValidatorSet {
 
 	// MODIFIERS
 	modifier onlySystemAndNotFinalized() {
-		require(msg.sender != SYSTEM_ADDRESS || finalized);
+		require((msg.sender == SYSTEM_ADDRESS) && !finalized);
 		_;
 	}
 
 	modifier whenFinalized() {
-		require(!finalized);
+		require(finalized);
 		_;
 	}
 
-	modifier isValidator(address _someone) {
```

Comment:
Maybe we can rename this modifier to `maybeValidator`. It's true that the function doesn't guarantee that the given address is a validator, but it guarantees that it's a validator or it's pending/waiting to be one.